libcloud/common/base.py

```diff
@@ -738,8 +748,15 @@ def request(self, action, params=None, data=None, headers=None,
 
                 self.connection.endheaders()
             else:
-                self.connection.request(method=method, url=url, body=data,
-                                        headers=headers)
+                if retry_enabled:
+                    retry(self.connection.request, self.timeout,
```

Comment:
I think what @Kami was trying to say, and I finally get it, is that if retry is a decorator, you can just use it inline here to create a new method that has the retry logic applied:

```
retrying_request = retry(self.connection.request, self.timeout, self.retry_delay, self.backoff)
retrying_request(method=method, url=url, body=data, headers=headers)
```

This will let you still override the values at runtime, but also allow retry to be used as a decorator elsewhere if people want to.  Does that make sense?